packages/pluggableWidgets/datagrid-web/src/Datagrid.tsx

```diff
@@ -25,13 +25,12 @@ export default function Datagrid(props: DatagridContainerProps): ReactElement {
     const multipleFilteringState = useMultipleFiltering();
     const { FilterContext } = useFilterContext();
 
-    props.datasource.requestTotalCount(true);
-
-    useState(() => {
+    useEffect(() => {
+        props.datasource.requestTotalCount(true);
```

Comment:
do we also want to do this everytime `props.pageSize` changes? 😄 
 
Do we haven have to do this more than once? If not, just put it inside its own useEffect with no deps

Reply:
It will probably just receive once, as this is a number.. so i dont see problems.